code/mod_table/mod_table.cpp

```diff
@@ -228,16 +228,36 @@ void parse_mod_table(const char *filename)
 		}
 
 		if (optional_string("$Splash screens:")) {
-			SCP_string splash_bitmap;
 			while (optional_string("+Bitmap:")) {
-				stuff_string(splash_bitmap, F_NAME);
+				splash_screen splash;
+				stuff_string(splash.filename, F_NAME);
 
 				// remove extension?
-				if (drop_extension(splash_bitmap)) {
-					mprintf(("Game Settings Table: Removed extension on splash screen file name %s\n", splash_bitmap.c_str()));
+				if (drop_extension(splash.filename)) {
+					mprintf(("Game Settings Table: Removed extension on splash screen file name %s\n", splash.filename.c_str()));
 				}
 
-				Splash_screens.push_back(splash_bitmap);
+				if (optional_string("+Aspect Ratio:")) {
+					stuff_float(&splash.aspect_ratio_exact);
+				}
+				if (optional_string("+Aspect Ratio Min:")) {
+					stuff_float(&splash.aspect_ratio_min);
+				}
+				if (optional_string("+Aspect Ratio Max:")) {
+					stuff_float(&splash.aspect_ratio_max);
+				}
+
+				if (splash.aspect_ratio_exact != 0.0f && (splash.aspect_ratio_min != 0.0f || splash.aspect_ratio_max != 0.0f)) {
+					Warning(LOCATION, "Game Settings Table: An exact aspect ratio and either a min or max aspect ratio were supplied for '%s'.  Only the exact value will be used.", splash.filename.c_str());
+					splash.aspect_ratio_min = 0.0f;
+					splash.aspect_ratio_max = 0.0f;
+				}
+
+				if (splash.aspect_ratio_exact == 0.0f && splash.aspect_ratio_min == 0.0f && splash.aspect_ratio_max == 0.0f) {
+					splash.is_default = true;
```

Comment:
Can we have a check to validate that there is at least one default splash screen available?
Not sure we should be relying on bm_load to throw an error if the default graphic isn't supplied with something like a TC that forgot to define a default splash.

Reply:
Splash screens are optional.  "Default" here just means that it will be a candidate if there are matches to the aspect ratio.  If no splash screen is applicable, the game falls back to `Game_title_screen_fname`.  If no file is loaded at all, `game_title_screen_draw` makes allowances for the bitmap being -1.

Reply:
Unless you mean the warning `"Could not open file 2_Preload!"` that would be shown.

Reply:
I'm not sure a full-fledged check makes sense.  A file check would need to duplicate some of the code that already exists in `bm_load`.  I could add a log statement though.

Reply:
Fair enough, I guess it's fine as is then